rflx/expression.py

```diff
@@ -314,6 +347,13 @@ def neutral_element(self) -> int:
     def symbol(self) -> str:
         return ' and then '
 
+    def z3expr(self) -> z3.BoolRef:
+        z3exprs = [t.z3expr() for t in self.terms]
+        boolexprs = [t for t in z3exprs if isinstance(t, z3.BoolRef)]
+        if len(z3exprs) != len(boolexprs):
+            raise ValueError
```

Comment:
Shouldn't that be also a `TypeError`? The actual issue is that one of the terms is not of the expected type. (Same for further occurrences below.)

> ValueError
> Raised when an operation or function receives an argument that has the right type but an inappropriate value [...]

Reply:
Good catch. Addressed in f067aa0.

Reply:
Also for all the other occurrences: f69c43e